src/test/lambda/utilities/getMainSourceFile.test.ts

```diff
@@ -342,4 +343,81 @@ describe('getMainSourceFile', async () => {
             assert.fail('Expected an exception, but none was thrown.')
         })
     })
+
+    describe('dotnet', async () => {
+        function createTestTemplate(): CloudFormation.Template {
+            return {
+                Resources: {
+                    HelloWorld: {
+                        Type: 'AWS::Serverless::Function',
+                        Properties: {
+                            Handler: 'HelloWorld::HelloWorld.Function::FunctionHandler',
+                            CodeUri: './artifacts/HelloWorld.zip',
+                            Runtime: 'dotnet'
+                        }
+                    }
+                }
+            }
+        }
+        const expectedHandlerPath = path.join(path.sep, 'dir', 'src', 'HelloWorld', 'Program.cs')
+        it('returns the URI of the main source file for a valid template', async () => {
+            const actual: vscode.Uri = await getMainSourceFileUri({
+                root: vscode.Uri.file('/dir'),
+                getLocalTemplates: (...workspaceUris: vscode.Uri[]) => toAsyncIterable([
+                    vscode.Uri.file('/dir/template.yaml')
+                ]),
+                loadSamTemplate: async uri => createTestTemplate(),
+                fileExists: async p => path.extname(p) === '.cs'
+            })
+
+            assert.strictEqual(actual.fsPath, expectedHandlerPath)
+        })
+
+        it('recognizes all Dotnet runtimes', async () => {
+            async function test(runtime?: string): Promise<void> {
+                const templateUri = vscode.Uri.file(path.join('/dir', 'template.yaml'))
+                const actual: vscode.Uri = await getMainSourceFileUri({
+                    root: vscode.Uri.file('/dir'),
+                    getLocalTemplates: (...workspaceUris: vscode.Uri[]) => toAsyncIterable([templateUri]),
+                    loadSamTemplate: async uri => {
+                        const template = createTestTemplate()
```

Comment:
`createTestTemplate` could have a `runtime` arg that sets this, and defaults to `dotnet`

Reply:
✔️